mozci/task.py

```diff
@@ -358,6 +360,30 @@ def configuration(self):
         parts = config.split("-")
         return "-".join(parts[:-1] if parts[-1].isdigit() else parts)
 
+    @property
+    def overhead(self):
```

Comment:
We discussed this, but I don't recall what we decided.
Should we:
1) consider overhead as the setup/teardown of the task (which this function is calculating);
2) consider overhead as the setup/teardown of the task + the amount of time elapsed between groups;
3) consider the setup/teardown overhead and overhead between groups separately.

CC @ahal @klahnakoski

Reply:
This is the definition of overhead I used after discussing with @klahnakoski on Matrix.

I originally used the definition of 2. However, this ran into a problem with particular task `JCAGLvN0S7-XbnjhSmocyg` and possibly `xpcshell` as a whole.

The problem being, in that particular task, test manifests are run in parallel. So, taking the definition of 2, I would have calculated:

`task_duration - sum([group_end - group_start for group in task])`

This produced a total group runtime of somewhere between 5000-7000, depending on how the query was structured.
However, the `task_duration` of that particular task was 1861, so taking the difference resulted in a large negative value as the overhead.

With the definition of 1, it avoids the problem altogether, but we lose insight to what the inter-group overhead might be.

I put this patch up after suggestion that we have something that we can work with then iterate on it. Also because my last day prior to parental leave is tomorrow.

Reply:
@marco-c   I would prefer to go with `overhead` defined as

> our best estimate of the setup/teardown of the task + the amount of time elapsed between groups

Which makes the current algorithm an underestimate, and we can improve it later by including the inter group run times.

Reply:
Agree going with 1) for now is easiest. We can iterate on it later to solve things like parallel execution. May need to update harnesses to adjust log messages to solve it.